### PR TITLE
CASSANDRA-13666 Avoid rendering static 2i entries as stales and removing them

### PR DESCRIPTION
**Pull request by bereng**

### CODE REVIEW

**Comment by bereng** [comment]
[CI run](https://circleci.com/workflow-run/bf02e9ad-a1e4-4700-bfba-daed55fcfc3c) passes utests which are the main ones to worry about. Failing dtests are due to connection issues, I tested some locally and they pass

File: src/java/org/apache/cassandra/index/internal/composites/CompositesSearcher.java [hunk 1]
```diff
@@ -231,16 +231,27 @@ private IndexEntry findEntry(Clustering clustering)
...
                     // tombstone or partition level deletion. Delete such stale entries.
-                    int cmp = comparator.compare(entry.indexedEntryClustering, clustering);
+                    int cmp = comparator.compare(indexedEntryClustering, clustering);
                     assert cmp <= 0; // this would means entries are not in clustering order, which shouldn't happen
                     if (cmp == 0)
                         return entry;
                     else
-                        staleEntries.add(entry);
+                    {
+                        boolean isStaticClustering = dataIter.metadata().hasStaticColumns();
```
Comment (bereng) [inline src/java/org/apache/cassandra/index/internal/composites/CompositesSearcher.java]: Might read a bit awkward, but given I am in a hot path I wanted to incur the cost to test for nulls only _if_ needed

Comment (blerer) [reply src/java/org/apache/cassandra/index/internal/composites/CompositesSearcher.java]: It is effectively a bit hard to read. I would have done something like:

```
boolean hasStaticColumns = dataIter.metadata().hasStaticColumns();

 // COMPACT COMPOSITE tables support null values in there clustering key but
 // those tables do not support static columns. By consequence if a table
 // has some static columns and all its clustering key elements are null
 // it means that the partition exists and contains only static data 
if (!hasStaticColumns() || !containsOnlyNullValues(indexedEntryClustering) 
    staleEntries.add(entry);
```
What do you think?

Comment (bereng) [reply src/java/org/apache/cassandra/index/internal/composites/CompositesSearcher.java]: @blerer if you give me thumbs up on the latest commit I just did I'll rebase, squash, forward merge and run CI before handing it over to you for committing.

**Comment by bereng** [comment]
@blerer squahsed, rebased & forward merged. Thx!

**Pull request closed without merging**